### PR TITLE
Improvements for auto-select opt-in mode

### PR DESCRIPTION
**Pull request by ThomasMichon**
### Overview

This change adds non-breaking improvements to #4907:

1. Pass `selectionZoneProps` through `DetailsList` to avoid having to pass each prop individually.
2. Add `data-selection-auto-select` as a preferred alias for `data-selection-select` to match original proposal in #4866.

### CODE REVIEW

File: packages/office-ui-fabric-react/src/utilities/selection/SelectionZone.test.tsx [hunk 1]
```diff
@@ -49,7 +49,7 @@ function _initializeSelection(selectionMode = SelectionMode.multiple): void {
         <button id='toggle1' data-selection-toggle={ true }>Toggle</button>
         <button id='invoke1' data-selection-invoke={ true }>Invoke</button>
         <button id='noSelect1'>No Select</button>
-        <button id='select1' data-selection-select={ true }>Select First</button>
+        <button id='select1' data-selection-auto-select={ true }>Select First</button>
```
Comment (dzearing) [inline packages/office-ui-fabric-react/src/utilities/selection/SelectionZone.test.tsx]: this is too verbose. Why this change?

File: packages/office-ui-fabric-react/src/components/DetailsList/DetailsList.tsx [hunk 2]
```diff
@@ -392,6 +392,7 @@ export class DetailsList extends BaseComponent<IDetailsListProps, IDetailsListSt
                 onItemInvoked={ onItemInvoked }
                 onItemContextMenu={ onItemContextMenu }
                 enterModalOnTouch={ this.props.enterModalSelectionOnTouch }
+                { ...(selectionZoneProps || {}) }
```
Comment (JasonGore) [inline packages/office-ui-fabric-react/src/components/DetailsList/DetailsList.tsx]: Is `{}` needed? Spreading undefined should be ok

File: packages/office-ui-fabric-react/src/utilities/selection/SelectionZone.tsx [hunk 3]
```diff
@@ -196,7 +197,10 @@ export class SelectionZone extends BaseComponent<ISelectionZoneProps, {}> {
           break;
         } else if (this._hasAttribute(target, SELECTION_INVOKE_ATTRIBUTE_NAME)) {
           break;
-        } else if ((target === itemRoot || this._hasAttribute(target, SELECTION_SELECT_ATTRIBUTE_NAME))
+        } else if (
+          (target === itemRoot ||
+            this._hasAttribute(target, SELECTION_AUTO_SELECT_ATTRIBUTE_NAME) ||
+            this._hasAttribute(target, SELECTION_SELECT_ATTRIBUTE_NAME))
```
Comment (JasonGore) [inline packages/office-ui-fabric-react/src/utilities/selection/SelectionZone.tsx]: Should this check for the new attribute be in _onKeyDown too?

File: packages/office-ui-fabric-react/src/utilities/selection/SelectionZone.test.tsx (hunk 1 above)
Comment (JasonGore) [reply packages/office-ui-fabric-react/src/utilities/selection/SelectionZone.test.tsx]: I also think `selection-select` is redundant with no obvious reasoning, though
